packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -171,6 +171,18 @@ const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
     marginBottom: 'none',
   };
 
+  /**
+   * Validate if the Menu Items has Id and Title filled
+   *
+   * @param {Array} menuItems array of Items
+   * @returns {Array} filtered array of items
+   */
+  const validateMenuItems = menuItems => {
+    return menuItems.filter(
+      item => item.title.length > 0 && item.id.length > 0
+    );
+  };
+
```

Comment:
@RaphaelAmadeu I feel like this might be a bit overkill, and we probably if anything would like to lean more towards properly defining the PropTypes, as well as `defaultProps` as much as possible. The warnings would come in through the console, and we're essentially leaning on prop-types to do its thing instead of building extra guard rails like this in all of our patterns.

Reply:
@RaphaelAmadeu I tend to agree with @jeffchew. I've thought about this a lot myself, and I think letting prop-types handle this is probably the way to go for now. Unless I'm missing something and this isn't simply props validation?

Reply:
@jeffchew , @kennylam ,

I thought that only the Proptype to be set required would prevent this from happening, but if an empty string is provided, Proptypes don't point it as an issue.